modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java

```diff
@@ -88,11 +88,7 @@ public int doStartTag() throws JspException {
 
 			if ((cpSku != null) && !hasChildCPDefinitions) {
 				_cpInstanceId = cpSku.getCPInstanceId();
-				_disabled =
-					!cpSku.isPurchasable() ||
-					((_commerceAccountId <= 0) &&
-					 !_commerceOrderHttpHelper.isGuestCheckoutEnabled(
-						 httpServletRequest));
+				_disabled = !cpSku.isPurchasable() || (_commerceAccountId == 0);
```

Comment:
I think even in this case the button will be disabled for guest user. At this point I think the code should be like this:
````suggestion
_disabled = !cpSku.isPurchasable()
````

Reply:
accountId 0 means that the user is signed in but it didn't select an account. So, I think it is correct to disable it in this scenario.

Reply:
@andrea-ale-sbarra , just to give context in B2C, a personal account is created as soon as user logs in. accountId will never be 0 in this case. In B2B, only time accountId 0 is user is signed in, but it did not select an account. acountId will always be -1 if Guest user: https://github.com/liferay-commerce/liferay-portal/blob/f81bedd689912175457afaef6b78108a50309007/modules/apps/account/account-api/src/main/java/com/liferay/account/constants/AccountConstants.java#L26

Reply:
Oh yeah you're right IDK why I was convinced that GUEST was 0 and not -1

Thank you for the explanations!